core/src/parse.rs

```diff
@@ -125,9 +125,15 @@ fn stack_ident(
 }
 
 /// Parses a `LangValue`, like "10" or "-3", not including any surrounding
-/// whitespace. (Negatives don't actually work yet).
+/// whitespace.
 fn lang_value(input: &str) -> IResult<&str, LangValue, VerboseError<&str>> {
-    map_res(digit1, |s: &str| s.parse::<LangValue>())(input)
+    let (input, is_neg) = opt(char('-'))(input)?;
+    let (input, val) =
+        map_res(digit1, |s: &str| s.parse::<LangValue>())(input)?;
+    match is_neg {
+        Some(_) => Ok((input, -val)),
+        None => Ok((input, val)),
+    }
```

Comment:
I think there's a small bug here: I don't think we'll be able to parse the min value (`-2147483648`). Since it'll try to parse `2147483648` to an `i32`, which isn't valid. This is fine for now, but can you add tests that include `LangValue::MIN` and `LangValue::MAX`, and just comment out the `MIN` one so we remember to fix it?

Reply:
got it to work for mins